warn/warn_operation.go

```diff
@@ -108,3 +111,85 @@ func integerDivisionWarning(f *build.File) []*LinterFinding {
 	})
 	return findings
 }
+
+func stringEscapeWarning(f *build.File) []*LinterFinding {
+	var findings []*LinterFinding
+
+	build.WalkPointers(f, func(expr *build.Expr, stack []build.Expr) {
+		str, ok := (*expr).(*build.StringExpr)
+		if !ok {
+			return
+		}
+
+		var quotes, value string
+		if str.TripleQuote {
+			quotes = str.Token[:3]
+			value = str.Token[3 : len(str.Token)-3]
+		} else {
+			quotes = str.Token[:1]
+			value = str.Token[1 : len(str.Token)-1]
+		}
+
+		var problems []int // positions of the problems
+
+		escaped := false
+		// This for-loop doesn't correclty check for a backlash at the end of the string literal, but
+		// such string can't be parsed anyway, neither by Bazel nor by Buildifier.
+		for i, ch := range value {
+			if !escaped {
+				if ch == '\\' {
+					escaped = true
+				}
+				continue
+			}
+
+			switch ch {
+			case '\n', '\\', 'n', 'r', 't', 'x', '\'', '"', '0', '1', '2', '3', '4', '5', '6', '7':
```

Comment:
Where does the list come from?
Can you include a link?

Reply:
Done.